pybamm/parameters/thermal_parameters.py

```diff
@@ -118,59 +55,178 @@ def _set_dimensional_parameters(self):
         self.T_init_dim = pybamm.Parameter("Initial temperature [K]")
 
         # Planar (y,z) thermal diffusion timescale
-        self.tau_th_yz = self.rho_eff_dim * (self.geo.L_z ** 2) / self.lambda_eff_dim
+        self.tau_th_yz = (
+            self.rho_eff_dim(self.T_ref)
+            * (self.geo.L_z ** 2)
+            / self.lambda_eff_dim(self.T_ref)
+        )
 
     def T_amb_dim(self, t):
         "Dimensional ambient temperature"
         return pybamm.FunctionParameter("Ambient temperature [K]", {"Times [s]": t})
 
-    def _set_dimensionless_parameters(self):
-        "Defines the dimensionless parameters"
+    def rho_cn_dim(self, T):
+        "Negative current collector density [kg.m-3]"
+        inputs = {
+            "Temperature [K]": self.Delta_T * T + self.T_ref,
```

Comment:
for consistency with other functions in this script, dimensional functions should take dimensional temperature (i.e. convert from T to T_dim in the dimensionless function then pass T_dim to this function)

Reply:
Done (I think, please double check when I push).